node/src/blockchain/blockchain_bridge.rs

```diff
@@ -1517,128 +1810,114 @@ mod tests {
+            block_number: start_block,
+            from: wallet,
+            wei_amount: 20_000,
+        };
+        let retrieved_blockchain_transactions = RetrievedBlockchainTransactions {
+            new_start_block: start_block,
+            transactions: vec![blockchain_transaction],
+        };
         let mut subject = BlockchainBridge::new(
-            Box::new(BlockchainInterfaceMock::default()),
-            Box::new(PersistentConfigurationMock::new()),
+            Box::new(
+                BlockchainInterfaceMock::default()
+                    .retrieve_transactions_result(Ok(retrieved_blockchain_transactions)),
+            ),
+            Box::new(persistent_config),
             false,
         );
         let system = System::new("test");
-        subject.scan_error_subs_opt = Some(accountant.start().recipient());
-        let retrieve_transactions = RetrieveTransactions {
-            recipient: make_wallet("somewallet"),
-            response_skeleton_opt: None,
-        };
-
-        subject.handle_scan(
-            failure_handler,
+        let accountant_addr = accountant
+            .system_stop_conditions(match_every_type_id!(ScanError))
+            .start();
+        subject.received_payments_subs_opt = Some(accountant_addr.clone().recipient());
+        subject.scan_error_subs_opt = Some(accountant_addr.recipient());
+        subject.handle_scan_future(
+            BlockchainBridge::handle_retrieve_transactions,
             ScanType::Receivables,
             retrieve_transactions,
         );
 
-        System::current().stop();
         system.run();
         let accountant_recording = accountant_recording_arc.lock().unwrap();
-        let message = accountant_recording.get_record::<ScanError>(0);
-        assert_eq!(
-            message,
-            &ScanError {
-                scan_type: ScanType::Receivables,
-                response_skeleton_opt: None,
-                msg: "My tummy hurts".to_string()
-            }
-        );
-        assert_eq!(accountant_recording.len(), 1);
-        TestLogHandler::new().exists_log_containing("WARN: BlockchainBridge: My tummy hurts");
+        let msg_opt = accountant_recording.get_record_opt::<ScanError>(0);
+        assert_eq!(msg_opt, None, "We didnt expect a scan error: {:?}", msg_opt);
     }
 
     #[test]
-    fn handle_scan_handles_failure_with_skeleton() {
+    fn handle_scan_future_handles_failure() {
+        assert_handle_scan_future_handles_failure(RetrieveTransactions {
+            recipient: make_wallet("somewallet"),
+            response_skeleton_opt: Some(ResponseSkeleton {
+                client_id: 1234,
+                context_id: 4321,
+            }),
+        });
+
+        assert_handle_scan_future_handles_failure(RetrieveTransactions {
+            recipient: make_wallet("somewallet"),
+            response_skeleton_opt: None,
+        });
+    }
+
+    fn assert_handle_scan_future_handles_failure(msg: RetrieveTransactions) {
```

Comment:
**URGENCY STATUS: THIS IS A CASE OF CRIPPLING THE OLD TESTS, AND WE CANNOT TOLERATE LOSING THE TEST COVERAGE BECAUSE WE ARE IN HURRY. PLEASE FIX IT SO THAT BOTH SITUATIONS ARE TESTED** See the explanation above.

Reply:
Regarding the `handle_scan` method appropriated to the BlockchainBridge, three tests makes up the full coverage on the master branch. 

Those three are:
**handle_scan_handles_success** (the happy path test)
**handle_scan_handles_failure_without_skeleton** (a general failure while the skeleton is not there)
**handle_scan_handles_failure_with_skeleton** (a general failure while the skeleton was provided)

Now, how can you understand the skeleton? It's a construct which actually represents a request from a UI. It stores the two numbers that are important for a continued communication: client_id (represents the UI), and the context_id (represents the specific conversation).

Reply:
You miss the second one. I recommend to try grafting onto this method assert_handle_scan_future_handles_failure and you might find yourselves with two tests sharing this function as a test body thanks to which it will save some space.

Reply:
Hi Bert,

I'm a little confused by your comment.

It appears we are still testing all 3 cases.
We have created a helper function `assert_handle_scan_future_handles_failure` to help test both cases for the skeleton using test `handle_scan_future_handles_failure`.

Here the tests code.
```rust
    #[test]
    fn handle_scan_future_handles_failure() {
        assert_handle_scan_future_handles_failure(RetrieveTransactions {
            recipient: make_wallet("somewallet"),
            response_skeleton_opt: Some(ResponseSkeleton {
                client_id: 1234,
                context_id: 4321,
            }),
        });

        assert_handle_scan_future_handles_failure(RetrieveTransactions {
            recipient: make_wallet("somewallet"),
            response_skeleton_opt: None,
        });
    }
```